code/game/machinery/flatpacker.dm

```diff
@@ -0,0 +1,344 @@
+
+		update_appearance()
+		return ITEM_INTERACT_SUCCESS
+
+	return NONE
+
+/**
+ * Attempts to find the total material cost of a typepath (including our creation efficiency), modifying a list
+ * The list is modified as an assoc list: Material datum typepath = Cost
+ * If the type is found on a techweb, uses material costs from there
+ * Otherwise, the typepath is created in nullspace and fetches materials from the initialized one, then deleted.
+ *
+ * Args:
+ * type - Typepath of the item we are trying to find the costs of
+ * costs - Assoc list we modify and return
+ */
+/obj/machinery/flatpacker/proc/analyze_cost(type, costs)
+	var/comp_type = type
+	if(ispath(type, /datum/stock_part))
+		var/datum/stock_part/as_part = type
+		comp_type = initial(as_part.physical_object_type)
+		if(as_part.tier > print_tier)
+			print_tier = as_part.tier
+
+	var/by_techweb = !isnull(SSresearch.item_to_design[comp_type])
+	var/obj/item/null_comp = by_techweb ? null : new comp_type
+	var/list/mat_list = by_techweb ? SSresearch.item_to_design[comp_type][1].materials : null_comp.custom_materials
+	for(var/atom/mat as anything in mat_list)
+		var/mat_type = mat.type
+
+		CREATE_AND_INCREMENT(costs, mat_type, mat_list[mat] * inserted_board.req_components[type])
+
+	qdel(null_comp)
+	return costs
+
+/// Start building the currently inserted board, if possible
+/obj/machinery/flatpacker/proc/start_build()
+	. = FALSE
+	if(!inserted_board)
+		return
+	if(!materials.has_materials(needed_mats, creation_efficiency))
+		say("Not enough materials to begin production.")
+		return
+	if(print_tier > max_part_tier)
+		say("Design too complex.")
+		return
+	materials.use_materials(needed_mats, creation_efficiency)
+	playsound(src, 'sound/items/rped.ogg', 50, TRUE)
+	busy = TRUE
+
+	addtimer(CALLBACK(src, PROC_REF(finish_build), inserted_board), flatpack_time)
+	return TRUE
+
+/// turns the supplied board into a flatpack, and sets the machine as not busy
+/obj/machinery/flatpacker/proc/finish_build(board)
+	busy = FALSE
+	new /obj/item/flatpack(drop_location(), board)
+
+/obj/machinery/flatpacker/Exited(atom/movable/gone, direction)
+	. = ..()
+	if(gone == inserted_board)
+		inserted_board = null
+		needed_mats = null
+		print_tier = 1
+		update_appearance()
+
+/obj/machinery/flatpacker/ui_act(action, list/params)
+	. = ..()
+
+	if(.)
+		return
+
+	. = TRUE
+
+	switch(action)
+		if("build")
+			if(busy)
+				return FALSE
+			start_build()
+			return TRUE
+
+		if("ejectBoard")
+			inserted_board.forceMove(drop_location())
+			return TRUE
+
+		if("eject")
+			var/datum/material/ejecting = locate(params["ref"])
+			var/amount = text2num(params["amount"])
+			if(!isnum(amount) || !istype(ejecting))
+				return TRUE
+
+			materials.retrieve_sheets(amount, ejecting, drop_location())
+			return TRUE
+
+	return FALSE
+
+/obj/machinery/flatpacker/Destroy()
+	. = ..()
+	inserted_board = null // this could be destroyed but the relevant refactor isnt in yet
```

Comment:
```suggestion

/obj/machinery/flatpacker/Destroy()
	QDEL_NULL(inserted_board)
	. = ..()
```

You should delete it. The board gets ejected out when you deconstruct the machine anyway but if an admin deletes it then no side effects should be present